src/simsopt/geo/surfaceobjectives.py

```diff
@@ -420,3 +425,367 @@ def dJ_by_dsurfacecoefficients(self):
+        ddenom_by_dc = np.mean(0.5*dS_dc * B_QS[..., None]**2 + dS[..., None] * B_QS[..., None] * B_QS_dc, axis=(0, 1)) 
+        dJ_by_dc = (denom * dnum_by_dc - num * ddenom_by_dc) / denom**2 
+        return dJ_by_dc
+
+
+class Iotas(Optimizable):
+    """
+    This term returns the rotational transform on a boozer surface as well as its derivative
+    with respect to the coil degrees of freedom.
+    """
+
+    def __init__(self, boozer_surface):
+        Optimizable.__init__(self, x0=np.asarray([]), depends_on=[boozer_surface])
+        self.boozer_surface = boozer_surface
+        self.biotsavart = boozer_surface.bs 
+        self.recompute_bell()
+
+    def J(self):
+        if self._J is None:
+            self.compute()
+        return self._J
+    
+    @derivative_dec
+    def dJ(self):
+        if self._dJ is None:
+            self.compute()
+        return self._dJ
+
+    def recompute_bell(self, parent=None):
+        self._J = None
+        self._dJ_by_dcoefficients = None
+        self._dJ_by_dcoilcurrents = None
+    
+    def compute(self):
+        if self.boozer_surface.need_to_run_code:
+            res = self.boozer_surface.res
+            res = self.boozer_surface.solve_residual_equation_exactly_newton(tol=1e-13, maxiter=20, iota=res['iota'], G=res['G'])
+
+        self._J = self.boozer_surface.res['iota']
+        
+        booz_surf = self.boozer_surface
+        iota = booz_surf.res['iota']
+        G = booz_surf.res['G']
+        P, L, U = booz_surf.res['PLU']
+        dconstraint_dcoils_vjp = boozer_surface_dexactresidual_dcoils_dcurrents_vjp
+
+        # tack on dJ_diota = dJ_dG = 0 to the end of dJ_ds
+        dJ_ds = np.zeros(L.shape[0])
+        dJ_ds[-2] = 1.
+        adj = forward_backward(P, L, U, dJ_ds)
+
+        adj_times_dg_dcoil = dconstraint_dcoils_vjp(adj, booz_surf, iota, G, booz_surf.bs)
+        self._dJ = -1.*adj_times_dg_dcoil
+
+
+def boozer_surface_dexactresidual_dcoils_dcurrents_vjp(lm, booz_surf, iota, G, biotsavart):
+    r"""
+    For a given surface with points :math:`x` on it, this function computes the
+    vector-Jacobian product of:
+    
+    .. math::
+        \lambda^T \frac{d\mathbf{r}}{d\text{coils   }} &= [G\lambda - 2\lambda\|\mathbf B(\mathbf x)\| (\mathbf{x}_\varphi + \iota \mathbf{x}_\theta) ]^T \frac{d\mathbf B}{d\text{coils}} \\ 
+        \lambda^T \frac{d\mathbf{r}}{d\text{currents}} &= [G\lambda - 2\lambda\|\mathbf B(\mathbf x)\| (\mathbf{x}_\varphi + \iota \mathbf{x}_\theta) ]^T \frac{d\mathbf B}{d\text{currents}}
+    
+    where :math:`\mathbf{r}` is the Boozer residual.
+    G is known for exact boozer surfaces, so if G=None is passed, then that
+    value is used instead.
+
+    Args:
+        lm: adjoint variable,
+        booz_surf: boozer surface,
+        iota: rotational transform on the boozer surface,
+        G: constant on boozer surface,
+        biotsavart: biotsavart object (not necessarily the same as the one used on the Boozer surface). 
```

Comment:
When does it make sense for the biot savart to be different from the one in the boozer surface? also, where is set_points called?